open-sphere-base/core/src/main/java/io/opensphere/core/units/AbstractUnitsProvider.java

```diff
@@ -431,4 +451,22 @@ private void setPreferredUnits(String selectionLabel)
             setPreferredUnits(units);
         }
     }
+
+    /**
+     * Set the preferred units using the selection label.
+     *
+     * @param selectionLabel The long label.
+     */
+    public void setPrevPreferredUnits(Class<? extends T> units)
+    {
+        if (!Utilities.sameInstance(myPrevPreferredUnits.getAndSet(units), units))
+        {
+            if (myPreferences != null)

```

Comment:
Combine these two ifs into one line.

Reply:
separate for the notify call to go even if they chose the same units

Reply:
Notify call is still outside both ifs.